src/utilities/Dealer.cpp

```diff
@@ -64,35 +64,37 @@ std::vector<vector<double>> Dealer::getScoreMatrix(const std::vector<v::RobotVie
         // Calculate the score for each robot for a role; the row
         for (auto robot : allRobots) {
             double distanceScore{};
-            double flagScore{};
 
             if (stpInfoMap.find(roleName) != stpInfoMap.end()) {
                 // The shorter the distance, the lower the distance score
                 distanceScore = getScoreForDistance(stpInfoMap.find(roleName)->second, robot);
             }
 
             // The better the flags, the lower the score
-            flagScore = scoreForFlags(dealerFlags, robot);
-            row.push_back(distanceScore + flagScore);
+            auto flagScore = scoreForFlags(dealerFlags, robot);
+            row.push_back((distanceScore + flagScore.first)/(flagScore.second+1));
         }
         scores.push_back(row);
     }
     return scores;
 }
 
 // Calculate the score for all flags for a role for one robot
-double Dealer::scoreForFlags(const std::vector<Dealer::DealerFlag> &dealerFlags, const v::RobotView &robot) {
+std::pair <double, double> Dealer::scoreForFlags(const std::vector<Dealer::DealerFlag> &dealerFlags, const v::RobotView &robot) {
     double robotScore = 0;
+    double totalFactor = 0;
     for (auto flag : dealerFlags) {
-        robotScore += getScoreForFlag(robot, flag);
+        auto ScoreForFlag = getScoreForFlag(robot, flag);
+        robotScore += ScoreForFlag.first;
+        totalFactor += ScoreForFlag.second;
+    }
+    return std::make_pair(robotScore,totalFactor);
     }
-    return robotScore;
-}
 
 // Get the score of one flag for a role for one robot
-double Dealer::getScoreForFlag(v::RobotView robot, Dealer::DealerFlag flag) {
-    double factor = getFactorForPriority(flag);
-    return factor * getDefaultFlagScores(robot, flag);
+std::pair <double, double> Dealer::getScoreForFlag(v::RobotView robot, Dealer::DealerFlag flag) {
+    double factor = 1/getFactorForPriority(flag);
+    return std::make_pair(factor * getDefaultFlagScores(robot, flag),factor);
```

Comment:
Most of the Dealer stuff in changed or fixed in the WIPDealer Pr. I'll be sure to document them better in that PR.

Note of the confusion: The back end is a bit more confusing, however the front end (Play design) makes sense now. I would have liked to split them as well but this would lead too much rewriting of existing code.